module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,57 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+
+    for (const auto& arr : bitmap) {
+        for (const auto& el : arr) {
+            if (el <= ' ')
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& grayScale) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width);
+    int count = 1, tmp;
+
+    for (auto row = grayScale.begin(); row != grayScale.end(); row++) {
+        for (auto column = row->begin(); column != std::prev(row->end()); column++) {
+            if (*(std::next(column)) != *column) {
```

Comment:
Complicated. You want to check if column will be != std::prev(row->end()). 
And also want check below std::next(column) why?

Reply:
because if I used a loop that increases the counter to the last element, I couldn't compare std :: next (column), because this next element is already out of the array.

Reply:
I know :) But you try to complicate the solution here. I know You will find simpler solution :)